pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -243,11 +246,16 @@ public boolean readCompacted() {
                 if (entry != null) {
                     int batchSize = batchSizes.getBatchSize(i);
                     int stickyKeyHash = getStickyKeyHash(entry);
+                    long[] ackSet = getCursorAckSet(PositionImpl.get(entry.getLedgerId(), entry.getEntryId()));
```

Comment:
batchIndexesAcks does it already include the number of acks?

Reply:
Yes, but it includes all the acked bit.